### PR TITLE
Rspec Test for Prototype Controller

### PR DESCRIPTION
**Pull request by r-ume**
## WHY
- Making sure that unexpected actions related to variables defined in prototypes_controller.rb will not happen before releasing this application.

## WHAT 
- added tests for all seven actions in `prototype_controller.rb`

- With User Login
  - GET #index 
    - assigns prototype variable to @prototype
    - expects the requested prototypes to be decorated with PaginatingDecorator
    - assigns the requested filter_type to normal
    - renders the index template 
  - GET #show
    - assigns the requested prototype to  @prototype
    - expects the requested prototypes to be decorated with PaginatingDecorator
    - assigns the requested comment to a new instance of comment
    - renders the :show template
  - GET #new 
    - assigns the requested prototype to a new instance of prototype
    - renders the :new template
  - POST #create
    - with valid attributes 
      - creates a new prototype
      - redirects to the root path
      - shows the successful flash message
    - with invalid attributes
      - does not create a new prototype
      - redirects to the root path
      - shows the unsuccessful flash message
  - GET #edit 
      - assigns the requested prototype to @prototype
      - expects the requested prototypes to be decorated with PrototypeDecorator
      - renders the :edit template
  - PATCH #update
      - with valid attribues
        - assigns the requested prototype to @prototype
        - expects the requested prototypes to be decorated with PrototypeDecorator
        - updates prototype
        - redirects to root path
        - shows flash message when updating prototype successfully
      - with invalid attributes
        - assigns the requested prototype to @prototype
        - expects the requested prototypes to be decorated with PrototypeDecorator
        - does not update prototype
        - renders the :edit template
  - DELETE #destroy
     - assigns the requested prototype to @prototype
     - deletes the prototype
     - redirects to the root path

- Without user login 
  - GET #new
    - redirects to the login page
  - POST #create
    - redirects to the login page
  - PATCH #update
    - redirects to the login page
  - DELETE #destroy
    - redirects to the login page

### CODE REVIEW

File: spec/controllers/prototypes_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,234 @@
+require 'rails_helper'
+
+describe PrototypesController, type: :controller do
+  # 忘れないようにメモ:
+  # letに ! をつけると、exampleが実行する前にcreateしてくれる
```
Comment (ryouzi) [inline spec/controllers/prototypes_controller_spec.rb]: そうですよ

File: spec/controllers/prototypes_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  # 定義した変数が呼ばれるタイミングではない 注意
+
+  # pending - テストが失敗すれば、成功でも失敗でもない 保留　として扱われる。
+  # skip    - テストの実行を止めたいときに使う
+  # xit     - example全体をskipさせる
+  # xdescribe, xcontext - グループ全体をスキップさせる
```
Comment (ryouzi) [inline spec/controllers/prototypes_controller_spec.rb]: ここらへんgood

**Comment by ryouzi** [comment]
LGTM

File: spec/controllers/prototypes_controller_spec.rb (hunk 2 above)
Comment (r-ume) [inline spec/controllers/prototypes_controller_spec.rb]: ありがとうございます！

**Comment by r-ume** [comment]
マージさせていただきます！

**Pull request merged**